File: MobileApp/gymly/lib/models/trainer_news.dart
class TrainerNews {
  final String id;
  final DateTime createDate;
  final String subjectId;
  final String title;
  final String content;

  TrainerNews(
    this.id,
    this.createDate,
    this.subjectId,
    this.title,
    this.content,
  );

  factory TrainerNews.fromJson(Map<String, dynamic> json) {
    return TrainerNews(
      json["id"] as String,
      DateTime.parse(json["createDate"]),
      json["subjectId"] as String,
      json["title"] as String,
      json["content"] as String,
    );
  }
}

File: Backend/Microservices/PostService/PostService.Domain/Entities/TrainerNews.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PostService.Domain.Entities;

public class TrainerNews
{
  [BsonId]
  [BsonRepresentation(BsonType.ObjectId)]
  public string Id { get; set; } = ObjectId.GenerateNewId().ToString();

  [BsonElement("CreateDate")]
  public DateTime CreateDate { get; set; } = DateTime.UtcNow;

  [BsonElement("SubjectId")]
  public string SubjectId { get; set; }

  [BsonElement("Title")]
  public string Title { get; set; }

  [BsonElement("Content")]
  public string Content { get; set; }
}
